src/axom/primal/tests/primal_knot_vector.cpp

```diff
@@ -0,0 +1,358 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*! 
+ * \file primal_knot_vector.cpp
+ * \brief This file tests primal's knot vector functionality
+ */
+
+#include "gtest/gtest.h"
+
+#include "axom/slic.hpp"
+
+#include "axom/primal/geometry/KnotVector.hpp"
+
+namespace primal = axom::primal;
+
+//------------------------------------------------------------------------------
+TEST(primal_knotvector, degree_constructor)
+{
+  const int degree = 2;
+  const int npts = 5;
+  primal::KnotVector<double> kvector(npts, degree);
+
```

Comment:
Should we also test that the number of spans matches expectations?

Reply:
I've added some tests of this in the knot insertion test, thanks!